### PR TITLE
datadog-agent: 7.50.3 -> 7.56.2

### LINKED ISSUE #351119: datadog-agent: incompatible with Go > 1.21, needs update
> Heads up that this PR breaks datadog-agent as re-introduces the issue in https://github.com/NixOS/nixpkgs/issues/299469

_Originally posted by @ryanrasti in https://github.com/NixOS/nixpkgs/issues/344622#issuecomment-2433332686_

maintainers: @thoughtpolice @domenkozar

### PR DESCRIPTION
**Pull request by Smona**
Updates datadog-related packages for Go 1.22 compatibility, and fixes a few issues. This should be backported to 24.11, where datadog-agent is currently marked as broken.

- Updated packages to the minimum version that supports Go 1.22
- Stripped out windows-only integration that showed up as broken in datadog UI (agentcrashdetect)
- Stripped out random test integration whose config is somehow included in the default configs
- Compiled packages with docker support by default. This will save a somewhat heavy build step for users who need docker support, while only adding 2.5% to the bin folder size (4MB).
- Update deprecated options for the process agent
- Set `dd_agent_bin` argument for the process agent to communicate with the main agent.

Superset of #356308

Resolves #351119 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/networking/dd-agent/datadog-agent.nix [hunk 1]
```diff
@@ -37,19 +38,20 @@ let
...
-  vendorHash = "sha256-Rn8EB/6FHQk9COlOaxm4TQXjGCIPZHJV2QQnPDcbRnM=";
+  vendorHash = if stdenv.isDarwin
+               then "sha256-3Piq5DPMTZUEjqNkw5HZY25An2kATX6Jac9unQfZnZc="
+               else "sha256-FR0Et3DvjJhbYUPy9mpN0QCJ7QDU4VRZFUTL0J1FSXw=";
 
   subPackages = [
     "cmd/agent"
     "cmd/cluster-agent"
     "cmd/dogstatsd"
-    "cmd/py-launcher"
```
Comment (Smona) [inline pkgs/tools/networking/dd-agent/datadog-agent.nix:52]: this command appears to have been removed

**Comment by PedroHLC** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 358927`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>datadog-agent</li>
    <li>datadog-process-agent</li>
  </ul>
</details>

**Review by PedroHLC — APPROVED** [review]
Tested running a `--help` with all binaries, which seems to work as expected.

**Comment by JohnRTitor** [comment]
@ofborg eval

File: pkgs/tools/networking/dd-agent/integrations-core.nix [hunk 2]
```diff
@@ -33,7 +33,7 @@
 #
 # [1]: https://github.com/DataDog/integrations-core
 
-{ pkgs, python, extraIntegrations ? {} }:
+{ pkgs, python, extraIntegrations ? { }, }:
```
Comment (JohnRTitor) [inline pkgs/tools/networking/dd-agent/integrations-core.nix:36]: Why are we passing `pkgs` here? We should import lib, fetchFromGitHub and others as needed. Current implementation makes it less overridable.

**Comment by JohnRTitor** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 358927`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>datadog-agent</li>
    <li>datadog-process-agent</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>datadog-agent</li>
    <li>datadog-process-agent</li>
  </ul>
</details>

**Comment by JohnRTitor** [comment]
@ofborg build datadog-agent datadog-process-agent

File: pkgs/tools/networking/dd-agent/integrations-core.nix (hunk 2 above)
Comment (Smona) [reply pkgs/tools/networking/dd-agent/integrations-core.nix]: it is a little unusual. i was thinking about refactoring the integrations but didn't end up figuring out a way to improve them.

**Pull request merged**